config/timewindow.py

```diff
@@ -0,0 +1,108 @@
+""" A window time frame, which can be relative to sunset and sunrise
+"""
+from datetime import datetime, timedelta
+import logging
+
+from astral import Location
+
+from ml_tools import tools
+
+
+class TimeWindow:
+    def __init__(self, start, end):
+        self.start = start
+        self.end = end
+        self.location = None
+        self.last_sunrise_check = None
+
+    def use_sunrise_sunset(self):
+        return self.start.is_relative or self.end.is_relative
+
+    def inside_window(self):
+        if self.use_sunrise_sunset:
+            self.update_sun_times()
+            if self.end.time < self.start.time:
+                return self.start.is_after() or self.end.is_before()
+        return self.start.is_after() and self.end.is_before()
+
+    def update_sun_times(self):
+        if not self.use_sunrise_sunset:
+            return
+
+        date = datetime.now().date()
+        if self.last_sunrise_check is None or date > self.last_sunrise_check:
+            sun_times = self.location.sun()
+            self.last_sunrise_check = date
+
+            if self.start.is_relative:
+                self.start.time = (
+                    sun_times["sunset"] + timedelta(seconds=self.start.offset_s)
+                ).time()
+            if self.end.is_relative:
+                self.end.time = (
+                    sun_times["sunrise"] + timedelta(seconds=self.end.offset_s)
+                ).time()
+            logging.info(
+                "start_rec is {} end_rec is {}".format(self.start.time, self.end.time)
+            )
+
+    def set_location(self, location_config):
+        self.location = Location()
+        lat, lng = location_config.get_lat_long(use_default=True)
+        self.location.latitude = lat
+        self.location.longitude = lng
+
+        self.location.altitude = location_config.altitude
+        self.location.timezone = tools.get_timezone_str(lat, lng)
+
```

Comment:
Rather than coupling this to `location_config` would it be better to just pass in lat, lon, altitude etc?